src/components/Dropdown/Dropdown.test.tsx

```diff
@@ -0,0 +1,523 @@
+          render(<Dropdown {...defaultProps} onChange={onChangeMock} />);
+          expect(onChangeMock).not.toHaveBeenCalled();
+
+          await user.click(screen.getByRole('combobox'));
+          await user.click(
+            screen.getByText(selectedOption.value, {
+              selector: '[role=option]',
+            }),
+          );
+          expect(onChangeMock).toHaveBeenCalledWith(
+            selectedOption.id,
+            selectedOption.value,
+          );
+        });
+      });
+
+      describe('With keyboard', () => {
+        it(`should call the passed down onChange callback with the selected option id and value on ${KeyCodes.ENTER} press`, async () => {
+          const onChangeMock = jest.fn();
+          const user = userEvent.setup();
+          const selectedOption = defaultProps.options[1]!;
+          render(<Dropdown {...defaultProps} onChange={onChangeMock} />);
+          expect(onChangeMock).not.toHaveBeenCalled();
+
+          await user.click(screen.getByRole('combobox'));
+          await user.keyboard(`{${KeyCodes.ARROW_DOWN}}`);
+          await user.keyboard(`{${KeyCodes.ENTER}}`);
+          expect(onChangeMock).toHaveBeenCalledWith(
+            selectedOption.id,
+            selectedOption.value,
+          );
+        });
+
+        it(`should call the passed down onChange callback with the selected option id and value on "${SPACE_SYMBOL}" symbol press`, async () => {
+          const onChangeMock = jest.fn();
+          const user = userEvent.setup();
+          const selectedOption = defaultProps.options[1]!;
+          render(<Dropdown {...defaultProps} onChange={onChangeMock} />);
+          expect(onChangeMock).not.toHaveBeenCalled();
+
+          await user.click(screen.getByRole('combobox'));
+          await user.keyboard(`{${KeyCodes.ARROW_DOWN}}`);
+          await user.keyboard(SPACE_SYMBOL);
+          expect(onChangeMock).toHaveBeenCalledWith(
+            selectedOption.id,
+            selectedOption.value,
+          );
+        });
+      });
+
+      it('should close the option list when an option is selected', async () => {
+        const user = userEvent.setup();
+        const selectedOption = defaultProps.options[1]!;
+        render(<Dropdown {...defaultProps} />);
+
+        await user.click(screen.getByRole('combobox'));
+        expect(screen.getByRole('combobox')).toHaveAttribute(
+          'aria-expanded',
+          'true',
+        );
+
+        await user.click(
+          screen.getByText(selectedOption.value, {
+            selector: '[role=option]',
+          }),
+        );
+        expect(screen.getByRole('combobox')).toHaveAttribute(
+          'aria-expanded',
+          'false',
+        );
+      });
+    });
+  });
+});
+
+describe('onKeyDown', () => {
+  const defaultParams: OnKeyDownParams = {
+    event: {
+      preventDefault: jest.fn(),
+      code: '',
+    } as unknown as KeyboardEvent<HTMLDivElement>,
```

Comment:
This is required because we are asserting that our created object is a `KeyboardEvent<HTMLDivElement>`. Typescript knows that this is not correct, because it is missing fields, so we tell it ,that the object is `unknown`, so that it accepts our assertion